src/PowerShellEditorServices/Workspace/WorkspaceFileSystemWrapper.cs

```diff
@@ -0,0 +1,324 @@
+        private bool _ignoreReparsePoints = false;
+
+        /// <summary>
+        /// Gets the maximum depth of the directories that will be searched
+        /// </summary>
+        public readonly int MaxRecursionDepth;
+
+        /// <summary>
+        /// Gets or sets the logging facility
+        /// </summary>
+        public ILogger Logger;
+
+        /// <summary>
+        /// Gets the directory where the factory is rooted. Only files and directories at this level, or deeper, will be visible
+        /// by the wrapper
+        /// </summary>
+        public DirectoryInfoBase RootDirectory
+        {
+            get { return _rootDirectory; }
+        }
+
+        /// <summary>
+        /// Creates a new FileWrapper Factory
+        /// </summary>
+        /// <param name="rootPath">The path to the root directory for the factory.</param>
+        /// <param name="recursionDepthLimit">The maximum directory depth.</param>
+        /// <param name="allowedExtensions">An array of file extensions that will be visible from the factory. For exampe [".ps1", ".psm1"]</param>
+        /// <param name="ignoreReparsePoints">Whether objects which are Reparse Points should be ignored. https://docs.microsoft.com/en-us/windows/desktop/fileio/reparse-points</param>
+        /// <param name="logger">An ILogger implementation used for writing log messages.</param>
+        public WorkspaceFileSystemWrapperFactory(String rootPath, int recursionDepthLimit, string[] allowedExtensions, bool ignoreReparsePoints, ILogger logger)
+        {
+            MaxRecursionDepth = recursionDepthLimit;
+            _rootDirectory = new WorkspaceFileSystemDirectoryWrapper(this, new DirectoryInfo(rootPath), 0);
+            _allowedExtensions = allowedExtensions;
+            _ignoreReparsePoints = ignoreReparsePoints;
+            Logger = logger;
+        }
+
+        /// <summary>
+        /// Creates a wrapped <see cref="DirectoryInfoBase" /> object from <see cref="System.IO.DirectoryInfo" />.
+        /// </summary>
+        internal DirectoryInfoBase CreateDirectoryInfoWrapper(DirectoryInfo dirInfo, int depth) =>
+            new WorkspaceFileSystemDirectoryWrapper(this, dirInfo, depth >= 0 ? depth : 0);
+
+        /// <summary>
+        /// Creates a wrapped <see cref="FileInfoBase" /> object from <see cref="System.IO.FileInfo" />.
+        /// </summary>
+        internal FileInfoBase CreateFileInfoWrapper(FileInfo fileInfo, int depth) =>
+            new WorkspaceFileSystemFileInfoWrapper(this, fileInfo, depth >= 0 ? depth : 0);
+
+        /// <summary>
+        /// Enumerates all objects in the specified directory and ignores most errors
+        /// </summary>
+        internal IEnumerable<FileSystemInfo> SafeEnumerateFileSystemInfos(DirectoryInfo dirInfo)
+        {
+            IEnumerable<FileSystemInfo> result = null;
+            try
+            {
+                result = dirInfo.EnumerateFileSystemInfos("*", SearchOption.TopDirectoryOnly);
```

Comment:
I think we stopped using the `Enumerate*` methods because of a MacOS issue.  @TylerLeonhardt you remember that? Something about too many file handles being open?

Reply:
Thus using GetFiles?

https://github.com/PowerShell/PowerShellEditorServices/blob/029a6ccfd889904494067c5e3e292f8ee07ca163/src/PowerShellEditorServices/Workspace/Workspace.cs#L323

Reply:
Well the good news is (I think) I could somewhat easily use the existing `GetFiles` etc style enumeration inside the wrapper if this is a concern.

Reply:
Just in case, lets go with that.

Reply:
Fixed.  Now using the GetFiles and GetDirectories methods